tests/system/Validation/ValidationTest.php

```diff
@@ -1324,4 +1330,17 @@ public function testNestedArrayThrowsException(): void
             'beneficiaries_accounts.account_2.purpose'       => 'The PURPOSE field must be at least 3 characters in length.',
         ], $this->validation->getErrors());
     }
+
+    public function testRuleWithLeadingAsterisk(): void
+    {
+        $data = [
+            ['foo' => 1],
+            ['foo' => null],
+        ];
+
+        $this->validation->setRules(['*.foo' => 'required'], ['1.foo' => ['required' => 'Required {field}']]);
```

Comment:
This is not related with this bug, but `['1.foo' => ['required' => 'Required {field}']]` seems tricky .

Reply:
You know, it turned out that the option you suggested does not work. The default error text is returned.

Reply:
The error text will be defined for the ``*.foo`` field, and the ``1.foo`` field is passed.